model/config.go

```diff
@@ -473,3 +475,67 @@ func (store *istioConfigStore) Policy(instances []*ServiceInstance, destination
 
 	return out
 }
+
+// RejectConflictingEgressRules rejects rules that have a domain which is equal to
+// a domain of some other rule.
+// According to Envoy's virtual host specification, no virtual hosts can share the same domain.
+// The following code rejects conflicting rules deterministically, by a lexicographical order -
+// a rule with a smaller key lexicographically wins.
+// Here the key of the rule is the key of the Istio configuration objects - see
```

Comment:
Please move this to validation.go.

Reply:
Please see my previous comment - I put this function in config.go since it is different from the functions in validation.go where validation of single configuration objects is performed. If you think it still should be in validation.go, I will copy that function and the test to validation.go/validation_test.go.